apollo-router/src/plugins/rhai/mod.rs

```diff
@@ -230,6 +222,24 @@ impl Plugin for Rhai {
         })
     }
 
+    fn router_service(&self, service: router::BoxService) -> router::BoxService {
+        const FUNCTION_NAME_SERVICE: &str = "router_service";
+        if !self.ast_has_function(FUNCTION_NAME_SERVICE) {
+            return service;
+        }
+        tracing::debug!("router_service function found");
+        let shared_service = Arc::new(Mutex::new(Some(service)));
+        if let Err(error) = self.run_rhai_service(
+            FUNCTION_NAME_SERVICE,
+            None,
+            ServiceStep::Router(shared_service.clone()),
+            self.block.load().scope.clone(),
+        ) {
+            tracing::error!("service callback failed: {error}");
```

Comment:
Does this give a good error message? If not then maybe tell the user what they should do to to fix their code.

Reply:
that is the same message we use on the supergraph, execution and subgraph levels. It includes the error reported by rhai (like syntax errors, etc)

Reply:
@Geal this hasn't been addressed, do we have a followup issue?

Reply:
this does not need to be addressed in this issue, as it does the same thing as other rhai services. We can open an issue to look in more detail at rhai error handling though